### PR TITLE
Multiple target groups

### PR DESCRIPTION
**Pull request by dynamike**
This PR adds support for associating an ECS service with multiple targets groups against a single ALB or NLB. I've also modified the helloworld go app to create 2 HTTP listeners and log requests as JSON to make testing and debugging this module a bit easier. I've also extended the testing in this module to validate three use cases of this module

* ECS service with no AWS load balancer 
* ECS service with a single ALB 
* ECS service with a single NLB

This is a breaking change, and upgrade notes can be found in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-* ECS service.
-* Task definition using `golang:alpine` (see below).
-* Configurable associations with Network Load Balancers (NLB) and Application Load Balancers (ALB).
+## Default container definition
 
 We create an initial task definition using the `golang:alpine` image as a way
 to validate the initial infrastructure is working: visiting the site shows
-a simple Go hello world page. We expect deployments to manage the container
+a simple Go hello world page listening on two ports (8080 and 8081). This is
+meant to get a proof up concept instance up and running and to help with testing
```
Comment (eeeady) [inline README.md:13]:
```suggestion
meant to get a proof of concept instance up and running and to help with testing
```

File: main.tf [hunk 2]
```diff
@@ -3,48 +3,88 @@ locals {
...
+  # container_ports, otherwise use no_lb_container_ports
+  default_container_ports = [8080, 8081]
+
+  # base64 encoded version of the helloworld go app
+  base64_encode_helloworld = base64encode(file("${path.module}/helloworld.go"))
+
+  # default container definition to be used with the helloworld go app included
+  # in this repo. It currently supports 2 HTTP listeners configured on
+  # environment variables PORT1 and PORT2 and simple JSON requests logs
+  default_container_definitions = jsonencode(
```
[the remark this answers is not in the document]
Comment (dynamike) [reply main.tf:32]: yeah, I think you're right.

File: README.md [hunk 3]
```diff
@@ -133,6 +164,50 @@ module "app_ecs_service" {
...
+```hcl
+module "app_ecs_service" {
+  source = "trussworks/ecs-service/aws"
+  ...
+  lb_target_groups =
+  [
+    {
+      container_port              = 8443
+      container_health_check_port = 8080
+      lb_target_group             = module.alb.arn
```
Comment (cblkwell) [inline README.md:203]: Your example here does not match what the actual variable definition calls for -- one or the other needs to be fixed.

File: README.md [hunk 4]
```diff
@@ -58,16 +65,42 @@ module "app_ecs_service" {
...
   associate_nlb          = true
   nlb_subnet_cidr_blocks = ["10.0.0.0/24", "10.0.1.0/24", "10.0.2.0/24"]
-  lb_target_group   = module.target_group.id
+
+  lb_target_groups =
+  [
+    {
+      container_port             = 8443
+      container_healthcheck_port = 8080
+      lb_target_arn              = module.nlb.arn
```
Comment (cblkwell) [inline README.md:77]: Also doesn't match variable definition.

File: README.md [hunk 5]
```diff
@@ -35,12 +37,17 @@ module "app_ecs_service" {
...
 
   associate_alb      = true
   alb_security_group = module.security_group.id
-  lb_target_group    = module.target_group.id
+  target_groups =
+  [
+    {
+      container_port             = 8443
+      container_healthcheck_port = 8443
+      lb_target_arn              = module.alb.arn
```
Comment (cblkwell) [inline README.md:48]: Doesn't match variable definition.

**Pull request merged**